### ISSUE TITLE
Why Gatsby don't end the process on GraphQL "conflicting field type"?

### ISSUE THREAD

**Issue by marlomgirardi** [gatsbyjs/gatsby#25421]
## Summary

By default, if any conflict happens with GraphQL types, it will be omitted by Gatsby and a message will be shown:
https://github.com/gatsbyjs/gatsby/blob/d847855cbedac25522947c36e67e59b6b3bcefc4/packages/gatsby/src/schema/infer/type-conflict-reporter.ts#L130-L139

In some cases, missing information might not be an acceptable thing.

Log Sample:
```
... ONGOING BUILD PROCESS BEFORE PROBLEM ...
warn There are conflicting field types in your data.

If you have explicitly defined a type for those fields, you can safely ignore this warning message.
Otherwise, Gatsby will omit those fields from the GraphQL schema.

If you know all field types in advance, the best strategy is to explicitly define them with the `createTypes` action and skip inference with the `@dontInfer`
 directive.
Profile.fieldName:
 - type: boolean
   value: false
 - type: string
   value: 'FIELD VALUE'
success building schema - 0.423s
... KEEP RUNNING BUILD AFTER PROBLEM ...
```

## Relevant information

My problem is that I have multiple sources that are not controlled by my team and if some change happens in the API (contentful fields, for instance) I would like the build to break.

If the build is run locally, there might not be a problem but when running in a CI/CD env we will not catch the problem until a user tells us about the missing information.

In summary,  I would prefer the build to break instead of just omitting the field.

**Comment by vladar** [comment]
Hi @marlomgirardi !

Technically it should break if you query this data somewhere (as the field you are trying to query doesn't exist). Are you saying that this doesn't happen for you?

If so then it sounds like a bug. Can you provide a [minimal reproduction][reproduction] so that we could take a look at this? This is a simplified example of the issue that makes it clear and obvious what the issue is and how we can begin to debug it.

Such reproduction should:

1. Have a conflicting field
2. Query this field in one of the pages / static queries
3. Don't break with both of the two conditions above

If you're up for it, we'd very much appreciate if you could [provide a minimal reproduction][reproduction] and we'll be able to take another look.

Thanks for using Gatsby! 💜

[reproduction]: https://gatsby.dev/reproduction

**Comment by marlomgirardi** [comment]
I've tried it for a bit and in the production build, it stops the process as expected. The problem was that the field was not being used.

For the develop build in the reproduction branch it stops with this error.
![image](https://user-images.githubusercontent.com/5932332/86411703-5aff3a80-bcb5-11ea-96bb-ab9c536635a0.png)

And in the project that I'm working on, don't have this error. I just need to check if it is on the config of webpack, eslint or somewhere else.

It probably is related to my configuration tho. Either way, as soon as I'm able to reproduce this problem I will put it here.

**Comment by TylerBarnes** [comment]
Hi @marlomgirardi !

I'm closing this issue due to no reproduction being posted within 7 days.
Once you have a minimal reproduction feel free to re-open this issue and we'll be happy to take another look. 💜 

Thank you!
